### ISSUE TITLE
'InputDecoration', has no unnamed constructor that takes no arguments

### ISSUE THREAD

**Issue by sk92129** [maheshj01/searchfield#196]
**Describe the bug**
latest version breaks the build for android

**To Reproduce**
Build for Android

[X ] By clicking this checkbox, I confirm I am using the latest version of the package found on [pub.dev/searchfield](https://pub.dev/packages/searchfield)

**Expected behavior**
this used to build and run before 1.2.1

**Actual behavior**

```dart  
../../../.pub-cache/hosted/pub.dev/searchfield-1.2.1/lib/src/input_decoration.dart:62:3: Error: The superclass, 'InputDecoration', has no unnamed constructor that takes no arguments.
  SearchInputDecoration({
  ^^^^^^^^^^^^^^^^^^^^^
../../../.pub-cache/hosted/pub.dev/searchfield-1.2.1/lib/src/input_decoration.dart:196:54: Error: The getter 'maintainHintHeight' isn't defined for the class 'SearchInputDecoration'.
 - 'SearchInputDecoration' is from 'package:searchfield/src/input_decoration.dart' ('../../../.pub-cache/hosted/pub.dev/searchfield-1.2.1/lib/src/input_decoration.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'maintainHintHeight'.
      maintainHintHeight: maintainHintHeight ?? this.maintainHintHeight,
                                                     ^^^^^^^^^^^^^^^^^^
```

**Screenshots**

**Code sample**

```dart  

 child: SearchField(
        focusNode: _focusNode,
        selectedValue: _initialSearchFieldListItem,
        onSearchTextChanged: (query) {
          final filteredSuggestions = widget.items
              .where((item) =>
                  item.name.toLowerCase().startsWith(query.toLowerCase()))
              .map((item) =>
                  SearchFieldListItem(item.name, item: item.value))
              .toList();
          return filteredSuggestions;
        },

```
</details>

**Additional context**

**Comment by maheshj01** [comment]
Hi @sk92129, Thanks for filing the issue.  I just verified on the latest stable 3.27 with searchfield: ^1.21 And I am build the apk.  What version of flutter are you using can you share the output of `flutter doctor -v`

**Comment by sk92129** [comment]
```dart  
[✓] Flutter (Channel stable, 3.24.4, on macOS 14.6 23G80 darwin-arm64, locale en-US)
    • Flutter version 3.24.4 on channel stable at /Users/myname/fvm/versions/3.22.2
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 603104015d (7 weeks ago), 2024-10-24 08:01:25 -0700
    • Engine revision db49896cf2
    • Dart version 3.5.4
    • DevTools version 2.37.3

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
    • Android SDK at /Users/myname/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Users/myname/Library/Java/JavaVirtualMachines/corretto-19.0.2/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment Corretto-19.0.2.7.1 (build 19.0.2+7-FR)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 16.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16C5032a
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.3+-79915917-b509.11)

[✓] IntelliJ IDEA Ultimate Edition (version 2024.3.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 83.0.4
    • Dart plugin version 243.23177

[✓] VS Code (version 1.95.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.102.0

[✓] Connected device (3 available)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 14.6 23G80 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 14.6 23G80 darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 131.0.6778.140
    ! Error: Browsing on the local area network for Sean iPhone. Ensure the device is unlocked and attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

This also occurs on the azure devops pipeline running on ubuntu 22 (ubuntu-22.04) where the JDK is 21, and the flutter SDK is 3.24.3 

After I changed the line to this:
  searchfield: 1.2.0

Then the build succeeds.

**Comment by maheshj01** [comment]
Thanks for that info @sk92129 `maintainHintHeight` is added to flutter stable 3.27
If you are not planning to upgrade to latest flutter, then you should be using `searchfield 1.2.0`, 
`searchfield 1.2.1` does not really have any fixes it is just compatible to latest stable 3.27

I will add this to release notes to clear the confusion

Hope this clarifies everything.

**Comment by maheshj01** [comment]
Closing as solved
